gridt/src/app/core/settings.service.spec.ts

```diff
@@ -93,32 +114,163 @@ describe("SettingsService when authentication fails", () => {
 
 describe("SettingsService when authentication is succesful", () => {
   beforeEach( () => {
+    apiStub = jasmine.createSpyObj('api', {'userIdentity$': mock_id[0]});
+
     secStoreStub = jasmine.createSpyObj(
       'secStore', {'get$': of(mock_id[0]), 'set$': of(true), 'clear$': of(true), 'remove$': of(true)}
     );
 
+    httpClientStub = jasmine.createSpyObj(
+      'httpClient', ['get', 'post', 'put']
+    );
+
     TestBed.configureTestingModule({
       providers: [
         SettingsService,
         {provide: AuthService, useClass: authServiceStub_succes},
+        {provide: ApiService, useValue: apiStub},
         {provide: SecureStorageService, useValue: secStoreStub},
+        {provide: HttpClient, useValue: httpClientStub}
       ]
     });
 
     service = TestBed.get(SettingsService as Type<SettingsService>);
-  })
+    api = TestBed.get(ApiService as Type<ApiService>);
+
+    let tick: (milliseconds: number) => void;
+    let fakeNow = 0;
+    tick = milliseconds => {
+      fakeNow += milliseconds;
+      tick(milliseconds);
+    };
+    asyncScheduler.now = () => fakeNow;
```

Comment:
In home.page.spec.ts or the swap.service.spec.ts file presents a way of setting the time in a more simple way if I understand the purpose of this piece of code correctly. If not, could you comment as to why you used this?

Reply:
I just looked around the internet for a solution. Will be happy to use a neater method.

Reply:
https://github.com/GridtNetwork/gridt-client/blob/5d1642df642846e64ccb7ae58e900ee62dc3158c/gridt/src/app/home/home.page.spec.ts#L24 

Does this what you want it to do?

Reply:
Well... I've tried that but it didn't work on the rxjs timer.

Reply:
Okey, it works now! Thanks for pointing me to the home.page.spec.ts example